core/src/main/resources/swagger/api-docs.yaml

```diff
@@ -5758,6 +5833,15 @@ components:
             Caller must have the add_project action for this service perimeter in
             Sam.
       description: ""
+    SpendReportConfiguration:
+      required:
+        - datasetName
```

Comment:
I leaned in this direction because I think it's just a bit simpler to deal with. It's also future-proof as you pointed out (although I don't know that we'd ever need it to be)

If I was going to take it in as a path parameter I would probably change the routes to something like:

`PUT /api/billing/v2/{projectId}/spendReportConfiguration/dataset/{datasetName}`
`DELETE /api/billing/v2/{projectId}/spendReportConfiguration/dataset`

Reply:
I think that makes sense. I forgot that the param is in the path. Feels like path params should be for things like entities in rawls as opposed to data being stored on an entity. That way, the API is called on the entity and has data passed in.